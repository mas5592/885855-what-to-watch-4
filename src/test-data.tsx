import {FilmType, ReviewType} from './types';

export const card: FilmType = {
  title: `Name`,
  genre: `film Genre`,
  date: 1993,
  coverBackground: `https://placeimg.com/1300/512/nature`,
  poster: `img/snatch.jpg`,
  picture: `img/snatch.jpg`,
  id: 123456,
  description: `Place to describe the movie`,
  rating: 8.9,
  scores: 240,
  director: `Director Name`,
  starring: [`Author`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  filmDurationTime: 107,
  videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  isFavorite: false,
  backgroundColor: `#FFFFFF`,
};

export const films: Array<FilmType> = [
  {
    title: `Name`,
    genre: `Genre`,
    date: 2014,
    coverBackground: `https://placeimg.com/1300/512/nature`,
    poster: `img/snatch.jpg`,
    picture: `img/snatch.jpg`,
    id: 123465,
    description: `Place to describe the movie`,
    rating: 8.3,
    scores: 1500,
    director: `Director Name`,
    starring: [`Author`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    filmDurationTime: 107,
    videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    isFavorite: false,
    backgroundColor: `#FFFFFF`,
  },
  {
    title: `Name`,
    genre: `Genre`,
    date: 2014,
    coverBackground: `https://placeimg.com/1300/512/nature`,
    poster: `img/snatch.jpg`,
    picture: `img/snatch.jpg`,
    id: 123478,
    description: `film description.`,
    rating: 8.3,
    scores: 1500,
    director: `Director Name`,
    starring: [`Author`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    filmDurationTime: 107,
    isFavorite: false,
    backgroundColor: `#FFFFFF`,
  }
];

export const activeFilm: FilmType = {
  title: `Snatch`,
  genre: `Genre`,
  date: 2014,
  coverBackground: `https://placeimg.com/1300/512/nature`,
  poster: `img/snatch.jpg`,
  picture: `img/snatch.jpg`,
  id: 123891,
  description: `film description.`,
  rating: 8.3,
  scores: 1500,
  director: `Director Name`,
  starring: [`Author`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  filmDurationTime: 107,
  videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  isFavorite: false,
  backgroundColor: `#FFFFFF`,
};

export const reviews: Array<ReviewType> = [{
  id: 1234567891,
  user: {
    id: 1223,
    name: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment.`,
  date: `2020-07-09T10:12:45.467Z`,
},
{
  id: 1234567891,
  user: {
    id: 1223,
    name: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment`,
  date: `2020-07-09T10:12:45.467Z`,
},
{
  id: 1234567891,
  user: {
    id: 1223,
    name: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment`,
  date: `2020-07-09T10:12:45.467Z`,
}];

export const genres: Array<string> = [
  `All genres`,
  `Drama`,
  `Thriller`,
  `Comedy`
];
