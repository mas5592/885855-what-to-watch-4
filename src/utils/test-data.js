export const card = {
  title: `Name`,
  genre: `film Genre`,
  date: 1993,
  coverBackground: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  picture: `img/snatch.jpg`,
  id: 123456,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege`,
  rating: 8.9,
  ratingDescription: `Very good`,
  scores: 240,
  director: `Director Name`,
  starring: [`Author`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/7/72/Landwasserviadukt%2C_aerial_video.webm`,
  filmDurationTime: 107,
  videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  isFavorite: false,
  backgroundColor: `#FFFFFF`,
};

export const films = [
  {
    title: `Snatch`,
    genre: `Genre`,
    date: 2014,
    coverBackground: `https://placeimg.com/1300/512/nature`,
    poster: `img/snatch.jpg`,
    picture: `img/snatch.jpg`,
    id: 123465,
    description: `film description.`,
    rating: 8.3,
    scores: 1500,
    director: `Director Name`,
    starring: [`Author`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    filmDurationTime: 107,
    videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    isFavorite: false,
    backgroundColor: `#FFFFFF`,
  }, {
    title: `Snatch`,
    genre: `Genre`,
    date: 2014,
    coverBackground: `https://placeimg.com/1300/512/nature`,
    poster: `img/snatch.jpg`,
    picture: `img/snatch.jpg`,
    id: 123478,
    description: `film description.`,
    rating: 8.3,
    scores: 1500,
    director: `Director Name`,
    starring: [`Author`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    filmDurationTime: 107,
    videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    isFavorite: false,
    backgroundColor: `#FFFFFF`,
  }
];

export const activeFilm = {
  title: `Snatch`,
  genre: `Genre`,
  date: 2014,
  coverBackground: `https://placeimg.com/1300/512/nature`,
  poster: `img/snatch.jpg`,
  picture: `img/snatch.jpg`,
  id: 123891,
  description: `film description.`,
  rating: 8.3,
  scores: 1500,
  director: `Director Name`,
  starring: [`Author`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  filmDurationTime: 107,
  videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  isFavorite: false,
  backgroundColor: `#FFFFFF`,
};

export const filmReviews = [{
  id: 1234567891,
  user: {
    id: 1223,
    title: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment`,
  date: `2020-07-09T10:12:45.467Z`,
},
{
  id: 1234567891,
  user: {
    id: 1223,
    title: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment`,
  date: `2020-07-09T10:12:45.467Z`,
},
{
  id: 1234567891,
  user: {
    id: 1223,
    title: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment`,
  date: `2020-07-09T10:12:45.467Z`,
}];

export const genres = [`All genres`, `Drama`, `Thriller`, `Comedy`];
