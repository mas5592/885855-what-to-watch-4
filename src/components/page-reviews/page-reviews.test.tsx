import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import PageReviews from './page-reviews';
import {reviews} from '../../test-data';
import NameSpace from '../../reducer/name-space';

const mockStore = configureStore([]);

describe(`PageReviews`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      filmReviews: reviews,
    },
  });

  it(`Should render correctly`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <PageReviews />
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
