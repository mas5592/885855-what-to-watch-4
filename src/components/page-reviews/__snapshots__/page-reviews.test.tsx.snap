// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageReviews Should render correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          My comment.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Ivan Ivanov
          </cite>
          <time
            className="review__date"
            dateTime="2020-07-09"
          >
            July 9, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8,0
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          My comment
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Ivan Ivanov
          </cite>
          <time
            className="review__date"
            dateTime="2020-07-09"
          >
            July 9, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8,0
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          My comment
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Ivan Ivanov
          </cite>
          <time
            className="review__date"
            dateTime="2020-07-09"
          >
            July 9, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8,0
      </div>
    </div>
  </div>
</div>
`;
